tests/test_file_save_gui.py

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) 2020 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import pytest
+from workfiles2_functions import _test_my_tasks_tab, _test_tab
+
+try:
+    from MA.UI import topwindows
+except ImportError:
+    pytestmark = pytest.mark.skip()
+
+
+@pytest.fixture(scope="module")
+def commands():
+    """
+    Return the command to launch Workfiles2 in different mode.
+    This fixture is used by the host_application fixture in conftest.py
+    """
+    return "file_save"
+
+
+@pytest.fixture(scope="module")
+def window_name():
+    """
+    Return the window app name.
+    This fixture is used by the app_dialog fixture in conftest.py
+    """
+    return "Shotgun: File Save"
+
+
+def test_my_tasks(app_dialog, sg_project):
+    """
+    Basic My Tasks tab UI validation to make sure all buttons, tabs and fields are available
+    """
+    # Validate that the right Workfiles 2 dialog mode is launched
+    assert app_dialog.root.captions["File Save"].exists(), "Not the File Open dialog"
+
+    # Validate File Save dialog buttons
+    assert app_dialog.root.buttons["Open"].exists(), "Open file type button is missing"
+    assert app_dialog.root.buttons["Save"].exists(), "Save button is missing"
+
+    # Validate File Save dialog text fields
+    assert app_dialog.root.textfields[
+        "Name Edit"
+    ].exists(), "Name text field is missing"
+    assert app_dialog.root.textfields[
+        "Version Number"
+    ].exists(), "Version text field is missing"
+
+    # Validate File Save dialog checkboxes
+    assert app_dialog.root.checkboxes[
+        "Use Next Available Version Number"
+    ].exists(), "Use Next Available Version Number checkbox is missing"
+    assert app_dialog.root.checkboxes[
+        "Use Next Available Version Number"
+    ].checked, "Use Next Available Version Number checkbox should be checked by default"
+
+    # My Tasks tab general UI validation
+    _test_my_tasks_tab(app_dialog, sg_project)
+
+
+# Parametrize decorator to run the same functions for Assets and Shots tabs.
+@pytest.mark.parametrize(
+    "tab_name, selection_hierarchy, entities",
+    [
+        (
+            "Assets",
+            ("Character", "AssetAutomation", "Model"),
+            ("Asset", "AssetAutomation", "Model", "Rig"),
```

Comment:
Perhaps a small comment on these explaining what they are and why the order is important?

I know I provided the selection_hierarchy as an example so I know that the order if the names of the tree items in the selection order, but it's not clear to others.
And the entities also appear to be in some kind of hierarchal order, but If I was to want to introduce another tab check for say Sequences then I'm not sure I'd know what to put here and what it is expecting. E.i. does it always have to be four items, like Entity Type. Entity Name, Step name, task name?

Are both model and rig tasks? If so then maybe it's work breaking those out into a tasks parameter?